iOSDevUK/Controllers/Sessions/ProgrammeTableViewController.swift

```diff
@@ -86,6 +98,41 @@ class ProgrammeTableViewController: UIViewController, UITableViewDelegate, UITab
         }
     }
     
+    /// Find the current or next session for the current set of sessions and scroll to it.
+    func scrollToCurrentSession(animated: Bool) {
+        guard let fetchedResultsController = fetchedResultsController,
+                let sections = fetchedResultsController.sections else {
+            return
+        }
+        
+        /// Find the session in the already-fetched results
+        let now = Date()
+        let nextOrCurrentSession = fetchedResultsController.fetchedObjects?.first { session in
+            guard let startTimeNSDate = session.startTime,
+                    let endTimeNSDate = session.endTime else {
+                return false
+            }
+            let startTime = startTimeNSDate as Date
+            let endTime = endTimeNSDate as Date
+            return (startTime <= now && endTime >= now) || (startTime >= now)
+        }
+        
+        /// If we found one, resolve it to a section index and row
+        if let sessionToScrollTo = nextOrCurrentSession {
+            for (sectionIndex, section) in sections.enumerated() {
+                if let rowIndex = section.objects?.firstIndex(where: {
+                    let session = $0 as! Session
+                    return session == sessionToScrollTo
+                }) {
+                    tableView.scrollToRow(at: IndexPath(row: rowIndex, section: sectionIndex), at: .top, animated: animated)
```

Comment:
Should we be using `.top` here or the default `.none`? `.none` would not scroll it if it's already visible e.g. at the bottom of the screen, it would only make sure that it's visible at all. I'm assuming that this is nicer, as it reduces unnecessary scrolling, but we'd probably have to try it out in the app to see how it feels.

Reply:
Yes, try it out. I did... it felt good - you just pull down if you wanted to check the previous session but... IMO a very rare case?